runner/runner.go

```diff
@@ -1421,13 +1422,59 @@ retry:
 		req.URL.Path = "/favicon.ico"
 		if faviconResp, favErr := hp.Do(req, httpx.UnsafeOptions{}); favErr == nil {
 			faviconMMH3 = fmt.Sprintf("%d", stringz.FaviconHash(faviconResp.Data))
-			builder.WriteString(" [")
-			if !scanopts.OutputWithNoColor {
-				builder.WriteString(aurora.Magenta(faviconMMH3).String())
+			if faviconMMH3 == "0" {
+				req.URL.Path = "/"
+				if faviconResp, favErr := hp.Do(req, httpx.UnsafeOptions{}); favErr == nil {
+					parsed, err := html.Parse(strings.NewReader(string(faviconResp.Data)))
+					var f func(*html.Node)
+					f = func(n *html.Node) {
```

Comment:
For better readability, I think this should be moved into an independent external function that performs the recursive call up to a max nested limit and return the hash to print to the output or error